### PR TITLE
Support customizing class label for torchmetrics.F1Score

### PR DESCRIPTION
**Pull request by zhiqiangdon**
`torchmetrics.F1Score` (https://torchmetrics.readthedocs.io/en/stable/classification/f1_score.html) doesn't provide the `pos_label` to return the f1 score of one specific class. However, sklearn's `f1_score` has this argument (https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html). In this PR, we define CustomF1Score to inherit from `torchmetrics.F1Score` and add the `pos_label` argument.

### CODE REVIEW

**Comment by sxjscience** [comment]
LGTM. The CI failure is not related to this PR.

**Review by bryanyzhu — COMMENTED** [review]
LGTM

**Comment by sxjscience** [comment]
Currently blocked by the CI but it looks good to me.

**Comment by szha** [comment]
Job PR-1840-9 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1840/9/index.html

**Review by sxjscience — APPROVED** [review]
LGTM
